src/pages/settings/tabs/currencies/edit-currency.vue

```diff
@@ -123,10 +142,40 @@ export default defineComponent({
       },
     );
 
+    onMounted(() => {
+      isChecked.value = props.currency.custom
+    })
+
     const onDeleteHandler = () => {
       emit('delete');
     };
 
+    const deleteExchangeRates = async () => {
+      try {
+        await Promise.all([
+          deleteCustomRate([
+            {
+              baseCode: props.currency.code,
+              quoteCode: props.currency.quoteCode,
+            },
+            {
+              baseCode: props.currency.quoteCode,
+              quoteCode: props.currency.code,
+            },
+          ]),
+            currenciesStore.loadCurrencies(),
+        ]);
+        emit('submit');
+
+        addSuccessNotification('Successfully updated.');
+      } catch (e) {
+        if (e.data.code === ERROR_CODES.validationError) {
+          addErrorNotification(e.data.message);
+          return;
+        }
+      }
```

Comment:
UI should always react even to unexpected errors. Add `addErrorNotification('Unexpected error')` at the end of the `catch` block.